src/Runner.Worker/StepsRunner.cs

```diff
@@ -458,10 +459,10 @@ private void CompleteStep(IStep step, IStep nextStep, TaskResult? result = null,
                         return;
                     }
 
+                    var parentScopeName = scope.ParentName;
```

Comment:
i dont think this is necessary. Code above should handle already. Outputs is special because leaving a scope. Outcome and conclusion should always be set already when step finishes.